src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,387 @@
+  {v : vector_measure α M} (hf₁ : ∀ i, measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) (hf₃ : ∀ i, v (f i) ≤ 0) :
+  v (⋃ i, f i) ≤ 0 :=
+(v.of_disjoint_Union hf₁ hf₂).symm ▸ tsum_nonpos hf₃
+
+lemma of_nonneg_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : 0 ≤ s A) (hB₂ : 0 ≤ s B)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+lemma of_nonpos_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : s A ≤ 0) (hB₂ : s B ≤ 0)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+end vector_measure
+
+namespace measure
+
+/-- A finite measure coerced into a real function is a signed measure. -/
+def to_signed_measure (μ : measure α) [hμ : finite_measure μ] : signed_measure α :=
+{ measure_of := λ i : set α, if measurable_set i then (μ.measure_of i).to_real else 0,
+  empty := by simp [μ.empty],
+  not_measurable := λ _ hi, if_neg hi,
+  m_Union :=
+  begin
+    intros _ hf₁ hf₂,
+    rw [μ.m_Union hf₁ hf₂, ennreal.tsum_to_real_eq, if_pos (measurable_set.Union hf₁),
+        summable.has_sum_iff],
+    { congr, ext n, rw if_pos (hf₁ n) },
+    { refine @summable_of_nonneg_of_le _ (ennreal.to_real ∘ μ ∘ f) _ _ _ _,
+      { intro, split_ifs,
+        exacts [ennreal.to_real_nonneg, le_refl _] },
+      { intro, split_ifs,
+        exacts [le_refl _, ennreal.to_real_nonneg] },
+        exact summable_measure_to_real hf₁ hf₂ },
+    { intros a ha,
+      apply ne_of_lt hμ.measure_univ_lt_top,
+      rw [eq_top_iff, ← ha, outer_measure.measure_of_eq_coe, coe_to_outer_measure],
+      exact measure_mono (set.subset_univ _) }
+  end }
+
+lemma to_signed_measure_apply_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : measurable_set i) :
+  μ.to_signed_measure i = (μ i).to_real :=
+if_pos hi
+
+/-- A measure is a vector measure over `ℝ≥0∞`. -/
+def to_ennreal_vector_measure (μ : measure α) : vector_measure α ℝ≥0∞ :=
+{ measure_of := λ i : set α, if measurable_set i then μ i else 0,
+  empty := by simp [μ.empty],
+  not_measurable := λ _ hi, if_neg hi,
+  m_Union := λ _ hf₁ hf₂,
+  begin
+    rw summable.has_sum_iff ennreal.summable,
+    { rw [if_pos (measurable_set.Union hf₁), measure_theory.measure_Union hf₂ hf₁],
+      exact tsum_congr (λ n, if_pos (hf₁ n)) },
+  end }
+
+@[simp]
+lemma to_ennreal_vector_measure_apply {μ : measure α}
+  {i : set α} (hi : measurable_set i) :
+  μ.to_ennreal_vector_measure i = μ i :=
+if_pos hi
+
+end measure
+
+namespace vector_measure
+
+variables {M : Type*} [add_comm_group M] [topological_space M]
+
+/-- The zero signed measure. -/
+def zero : vector_measure α M :=
+⟨0, rfl, λ _ _, rfl, λ _ _ _, has_sum_zero⟩
+
+instance : has_zero (vector_measure α M) := ⟨zero⟩
```

Comment:
I'm not sure the extra definition is worth anything here
```suggestion
/-- The zero signed measure. -/
instance : has_zero (vector_measure α M) := ⟨⟨0, rfl, λ _ _, rfl, λ _ _ _, has_sum_zero⟩⟩
```
If you want to keep it, make it `protected`.